src/main/java/thestonedturtle/mahoganyhomes/MahoganyHomesOverlay.java

```diff
@@ -64,6 +64,49 @@ class MahoganyHomesOverlay extends OverlayPanel
 		getMenuEntries().add(new OverlayMenuEntry(RUNELITE_OVERLAY, CLEAR_OPTION, "Contract"));
 	}
 
+	// Adds a line to the panel describing a material. type is the
+	// un-pluralized name of the material, min and max are the range
+	// required, and have is the number in the player's inventory.
+	void addMaterialLine(String type, int min, int max, int have) {
+		if (min == 0 && max == 0) {
+			return;
+		}
+		LineComponent.LineComponentBuilder line = LineComponent.builder();
+
+		if (min == max)
+		{
+			String plural = min > 1 ? "s" : "";
+			line = line.left(String.format("%d %s%s", min, type, plural));
+		} else {
+			// Always pluralize by adding an "s"
+			line = line.left(String.format("%d - %d %ss", min, max, type));
+		}
+
+		// If asked for, add a colored symbol on the right depicting if
+		// the player has enough supplies.
+		if (config.checkSupplies()) {
+			if (have >= max) {
+				line = line.right("✓").rightColor(Color.GREEN);
+			} else if (have >= min) {
+				// Player has between max and min, so it's not
+				// certain if they have enough.
+				line = line.right("?").rightColor(Color.YELLOW);
+			} else {
+				line = line.right("x").rightColor(Color.RED);
+			}
+		}
+		panelComponent.getChildren().add(line.build());
+	}
+
+	// Adds lines to the panel describing the materials required.
+	void addRequiredMaterials(RequiredMaterials mats) {
+		if (mats == null) {
+			return;
+		}
+
+		addMaterialLine("Plank", mats.MinPlanks, mats.MaxPlanks, plugin.getNumPlanksInInventory());
+		addMaterialLine("Steel Bar", mats.MinSteelBars, mats.MaxSteelBars, plugin.getNumSteelBarsInInventory());
```

Comment:
These `plugin.getNum...` functions will be called every time the overlay is redrawn, which is very often. Instead of recalculating these values every time the overlay is redrawn please cache these and update them when the inventory or plank sack contents change.

The easiest way to do this would be to add this to the `MahoganyHomesPlugin.java` file, update the existing functions to something like `calculateNum...` and then call them when necessary.
```java
	@Getter
	private int numPlanksInInventory = 0;
	@Getter
	private int numBarsInInventory = 0;
```